### PR TITLE
Allow numeric fields

### PR DESCRIPTION
**Pull request by jschwartzentruber**
A number of fields are numeric, but allow shortcuts like "1g" for 1 gigabyte, or "4h30m" instead of 16200 seconds. This allows casting any of those fields to string, if they are given numerically. This will also be useful by allowing `ENVVAR: 1` in yaml instead of requiring `ENVVAR: "1"`.

This includes a other few minor updates:
- Update README
- Fix timezone being ignored in `schedule_start`
- Add test for `schedule_start: null`
- Enable pycoverage calculation in pytest (not reported to codecov.io yet)

### CODE REVIEW

File: fuzzing_tc/common/pool.py [hunk 1]
```diff
@@ -23,13 +24,13 @@
...
-        "disk_size": str,
+        "cycle_time": (int, str),
+        "disk_size": (int, str),
         "imageset": str,
         "macros": dict,
-        "max_run_time": str,
+        "max_run_time": (int, str),
         "metal": bool,
-        "minimum_memory_per_core": str,
+        "minimum_memory_per_core": (float, str),
```
Comment (tysmith) [inline fuzzing_tc/common/pool.py:33]: Why is this `(float, str)` and not `(int, str)`?

Comment (jschwartzentruber) [reply fuzzing_tc/common/pool.py:33]: I thought it made sense to allow fractional RAM sizes, so you can do `"3.5g"` or whatever.  `disk_size` could also be `float`, but it seems unlikely to be used.

The other `int` fields are for `cores_per_task` and `tasks`, which make sense as whole numbers. `cycle_time` and `max_run_time` are in seconds, so don't need to be divided further.

Comment (tysmith) [reply fuzzing_tc/common/pool.py:33]: If you are already handling the string "3.5g" would it make sense to convert that value to MBs (since this is the smallest unit we will deal with) and store that value as an int?

Comment (jschwartzentruber) [reply fuzzing_tc/common/pool.py:33]: That's an easy change to make, however I think it makes it more error prone. If you missed the "g" it would default to 4MB instead of 4GB. The whole point of allowing the string suffixes is to avoid having to do math.
